specs/opcode/0AEXP.md

```diff
@@ -0,0 +1,59 @@
+# EXP opcode
+
+## EVM Behaviour
+
+The `EXP` opcode performs an exponential operation. It reads the integer base `base` and integer exponent `exponent` from the top of the stack and pushes the exponentiation result, i.e. `base ^ exponent (mod 2^256)` to the top of the stack.
+
+## Constraints
+
+1. opId == 0x0A
+2. State Transition:
+    - rw_counter += 3
+    - stack_pointer += 1
+    - pc += 1
+    - gas -= static_gas + dynamic_gas, where:
+        - `static_gas = 10`
+        - `dynamic_gas = 50 * byte_size(exponent)`
+3. Lookups:
+    - `base` is at the top of the stack (Read from RW Table)
+    - `exponent` is at the second position of the stack (Read from RW Table)
+    - `exponentiation` is at the top of the stack (Write to RW Table)
+    - if `exponent == 2`:
```

Comment:
There are no lookups in case `exponent == 0` or `exponent == 1`. In those cases, we validate that:
1. If `exponent == 0` then `exponentiation == 1`
2. If `exponent == 1` then `exponentiation == base`

Because there are no lookups, I did not mention in the gadget specs. It is mentioned in the `tables.md` though. Will add it here just to make it more verbose.